src/main/java/io/jsondb/crypto/Default1Cipher.java

```diff
@@ -0,0 +1,166 @@
+   */
+  public Default1Cipher(String base64CodedEncryptionKey, Charset charset) throws GeneralSecurityException {
+    this(Base64.getDecoder().decode(base64CodedEncryptionKey), charset);
+  }
+
+  /**
+   * Creates a new default cipher with the specified charset encoding and key.
+   * 
+   * @param encryptionKey  A symmetric key to be used during encryption and decryption.
+   * @param charset        The charset to be considered when encrypting plaintext or decrypting ciphertext.
+   */
+  public Default1Cipher(byte[] encryptionKey, Charset charset) throws GeneralSecurityException {
+    if (charset == null) throw new NullPointerException("charset");
+    if (encryptionKey == null) throw new NullPointerException("encryptionKey");
+    this.charset = charset;
+    this.key = new SecretKeySpec(encryptionKey, ENCRYPTION_ALGORITHM);
+  }
+
+  /** @{inheritDoc} */
+  @Override
+  public String encrypt(String plainText) {
+    try {
+      byte[] iv = new byte[IV_SIZE];
+      rnd.nextBytes(iv);
+      Cipher enc = Cipher.getInstance(CIPHER_ALGORITHM, PROVIDER);
+      enc.init(Cipher.ENCRYPT_MODE, key, new GCMParameterSpec(IV_SIZE * 8, iv));
+      byte[] input = plainText.getBytes(charset);
+      int sizeReq = IV_SIZE + enc.getOutputSize(input.length);
+      byte[] output = new byte[sizeReq];
+      ByteBuffer store = ByteBuffer.wrap(output);
+      store.put(iv);
+      int extra = enc.doFinal(input, 0, input.length, output, IV_SIZE);
+      store.position(store.position() + extra);
+      store.flip();
+      ByteBuffer bb = Base64.getEncoder().encode(store);
+      return new String(bb.array(), 0, bb.limit(), StandardCharsets.US_ASCII);
+    } catch (NoSuchPaddingException | NoSuchProviderException | NoSuchAlgorithmException | IllegalBlockSizeException |InvalidAlgorithmParameterException e) {
+      throw new JsonDBException("Default cipher cannot be used on this VM installation", e);
+    } catch (InvalidKeyException e) {
+      throw new JsonDBException("Invalid key", e);
+    } catch (BadPaddingException | ShortBufferException e) {
+      throw new JsonDBException("Unexpected (bug?) crypto error", e);
+    }
+  }
+
+  /** @{inheritDoc} */
+  @Override
+  public String decrypt(String cipherText) {
+    byte[] in = Base64.getDecoder().decode(cipherText);
+    try {
+      Cipher dec = Cipher.getInstance(CIPHER_ALGORITHM, PROVIDER);
+      dec.init(Cipher.DECRYPT_MODE, key, new GCMParameterSpec(IV_SIZE * 8, in, 0, IV_SIZE));
+      return new String(dec.doFinal(in, IV_SIZE, in.length - IV_SIZE), charset);
+    } catch (NoSuchPaddingException | NoSuchProviderException | NoSuchAlgorithmException | IllegalBlockSizeException e) {
+      throw new JsonDBException("Default cipher cannot be used on this VM installation", e);
+    } catch (InvalidKeyException e) {
+      throw new JsonDBException("Invalid key", e);
+    } catch (BadPaddingException | InvalidAlgorithmParameterException e) {
```

Comment:
On decryption with AEAD modes, one needs to handle  AEADBadTagException. This is thrown when the tag is invalid (which means the ciphertext was tampered with in some form, or the decryption key may have been incorrect).

AEADBadTagException is a subclass of BadPaddingException, but I think it should be handled separately and explicitly.